src/Provider/Here/Here.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Here;
+
+use Geocoder\Collection;
+use Geocoder\Exception\InvalidCredentials;
+use Geocoder\Exception\UnsupportedOperation;
+use Geocoder\Model\AddressBuilder;
+use Geocoder\Model\AddressCollection;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+use Geocoder\Http\Provider\AbstractHttpProvider;
+use Geocoder\Provider\Provider;
+use Geocoder\Provider\Here\Model\HereAddress;
+use Http\Client\HttpClient;
+
+/**
+ * @author Sébastien Barré <info@zoestha.de>
+ */
+final class Here extends AbstractHttpProvider implements Provider
+{
+    /**
+     * @var string
+     */
+    const GEOCODE_ENDPOINT_URL = 'https://geocoder.api.here.com/6.2/geocode.json?app_id=%s&app_code=%s&searchtext=%s&gen=8';
+
+    /**
+     * @var string
+     */
+    const REVERSE_ENDPOINT_URL = 'https://reverse.geocoder.api.here.com/6.2/reversegeocode.json?prox=%F,%F&250&app_id=%s&app_code=%s&mode=retrieveAddresses&gen=8&maxresults=%d';
+
+    /**
+     * @var string
+     */
+    private $appId = null;
+
+    /**
+     * @var string
+     */
+    private $appCode = null;
+
+    /**
+     * @param HttpClient $adapter An HTTP adapter.
+     * @param string     $appId   An App ID.
+     * @param string     $apoCode An App code.
+     */
+    public function __construct(HttpClient $client, $appId, $appCode)
```

Comment:
I would typehint the `$appId` and `appCode` parameters as `string`. Than we can remove the exception in the constructor.